src/dotty/tools/dotc/transform/Space.scala

```diff
@@ -288,14 +290,12 @@ class SpaceEngine(implicit ctx: Context) extends SpaceLogic {
     }
   }
 
-  // find the first redundant pattern and stop
-  def redundancy(_match: tpd.Match): Unit = {
+  def redundancy(_match: Match): Unit = {
     val Match(sel, cases) = _match
     val selTyp = widen(sel.tpe)
 
     // starts from the second, the first can't be redundant
-    var i = 1
-    while(i < cases.length) {
+    (1 until cases.length).foreach { i =>
```

Comment:
Ah, I see, you could do `cases.zipWithIndex.foreach { (c, i) => ... }`. There might be a better way but I can't think of it.